config/internal/configsource/manager.go

```diff
@@ -91,6 +110,22 @@ import (
 //      # as a string if params doesn't specify that "binary" is true.
 //      text_from_file: $file:/etc/text.txt
 //
+// Bracketed single-line should be used when concatenating a suffix to the value retrieved by
+// the config source. Example:
+//
+//    component:
+//      # Retrieves the value of the environment variable LOGS_DIR and appends /component.log to it.
+//      log_file_fullname: ${env:LOGS_DIR}/component.log
+//
+// Environment variables are expanded before passed to the config source when used in the selector or
+// the optional parameters. Example:
+//
+//    component:
+//      # Retrieves the value from the file text.txt located on the path specified by the environment
+//      # variable DATA_PATH. The name of the environment variable is the string after the delimiter
+//      # until the first character different than '_' and non-alpha-numeric.
+//      text_from_file: $file:$DATA_PATH/text.txt
```

Comment:
Can you also add description of 2 syntaxes we support for env variables here and describe how we differentiate between env variables and config source calls via `:` delimiter?

Reply:
Added more examples and explanations around it.